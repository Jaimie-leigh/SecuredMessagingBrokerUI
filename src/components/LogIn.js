// Inital screen displayed on load on the Broker Interface

// Options to improve:
// check pop up so be warning as text entered (like the length text does)
// on submit check username is in DB if not say username is not valid.

import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import "./Login.css";

export default function LogIn() {
  const history = useHistory();
  const [value, setValue] = useState("");
  const regexp = /^[0-9]{0,7}$/;
  const [isValid, setIsValid] = useState("");

  //sets value from username entered by broker
  // sets isValid to true or false if username is in expected format
  function handleChange(e) {
    setValue(e.target.value);
    if (!regexp.test(value)) {
      //alert("Username should not contain letters, please re-enter");
      setIsValid(false);
    } else {
      setIsValid(true);
    }
  }

  // on click of 'Log in' button checks username is in correct format
  // routes to broker landing page if correct
  function handleSubmit(e) {
    e.preventDefault();
    if (!isValid) {
      return;
    }
    return history.replace("/BrokerWelcome", value);
  }

  // renders on the page

  return (
    <div className="Login">
      <form onSubmit={handleSubmit}>
        <h1>Secured Messenging</h1>
        <label>
          Username:
          <input
            type="tel"
            name="BrokerId"
            value={value}
            onChange={handleChange}
            required
            maxLength="7"
            minLength="7"
          />
        </label>
        {isValid === false && (
          <div className="helpText">
            <p>
              Username should consist of 7 characters including only digitis,
              please re-enter your username
            </p>
          </div>
        )}
        <button>Log In</button>
      </form>
    </div>
  );
}
