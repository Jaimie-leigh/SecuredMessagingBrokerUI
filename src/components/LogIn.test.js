import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import {render, unmountComponentAtNode } from 'react-dom';
import LogIn from './LogIn';
import fireEvent from '@testing-library/react'

let container = null;

beforeEach(() => {
    // set a DOM elemant as render target
    container = document.createElement('div');
    document.body.appendChild(container);
})

afterEach(() => {
    // clean up on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

describe('Log in snapshot test', () => {
    test('test renders as per snapshot', () => {
        const tree = renderer
            .create(<LogIn></LogIn>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("test page renders as epxected when loading", () => {
        const searchInput = queryByLabelText('BrokerId');
    
        render(
          <ApplicationMessages location={{ state: state }}></ApplicationMessages>,
          container
        );

        fireEvent.change(searchInput, { target: { value: 'abcde' }})

        expect(container.innerHTML).toContain("numbers only allowed");
      });
});
