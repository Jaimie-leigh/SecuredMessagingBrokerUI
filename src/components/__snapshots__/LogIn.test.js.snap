// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log in snapshot test test renders as per snapshot 1`] = `
<div
  className="Login"
>
  <form
    onSubmit={[Function]}
  >
    <h1>
      Secured Messenging
    </h1>
    <label>
      Username:
      <input
        maxLength="7"
        minLength="7"
        name="BrokerId"
        onChange={[Function]}
        required={true}
        type="tel"
        value=""
      />
    </label>
    <button>
      Log In
    </button>
  </form>
</div>
`;
